### PR TITLE
Test : Pandas 1.0.5 → 1.1.0 behavior change on DataFrame.apply() where fn returns np.ndarray 

### LINKED ISSUE #35517: BUG: Pandas 1.0.5 → 1.1.0 behavior change on DataFrame.apply() where fn returns np.ndarray
- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the latest version of pandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

---

#### Code Sample, a copy-pastable example

```python
print(type(
  pd.DataFrame([['foo']])
  .apply(lambda col: np.array('bar'))
  .iloc[0]))
```

#### Output of Pandas 1.0.5

```
<class 'str'>
```

#### Output of Pandas 1.1.0

```
<class 'numpy.ndarray'>
```

It is not clear to me if this behaviour change is intended or not. I couldn't find anything obvious in the [release notes](https://pandas.pydata.org/pandas-docs/stable/whatsnew/v1.1.0.html).

I discovered it because it broke my code, which returns the output of `DataFrame.unique().squeeze()` (with the intent of extracting a scalar) in an `apply()` func. Arguably my code is wrong, because `DataFrame.unique()` returns an `np.ndarray`, and calling `squeeze()` on that always returns an ndarray, never a scalar - the correct call would have been `item()`. Up until now, Pandas would "hide" the bug because it would take the one-element ndarray and treat it as a scalar. Not anymore, it seems. My code ultimately fails downstream of the call as it tries to hash the resulting string, which doesn't work if the string is actually a 1-element ndarray.

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : d9fff2792bf16178d4e450fe7384244e50635733
python           : 3.6.9.final.0
python-bits      : 64
OS               : Linux
OS-release       : 4.19.104+
Version          : #1 SMP Wed Feb 19 05:26:34 PST 2020
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.1.0
numpy            : 1.19.1
pytz             : 2018.9
dateutil         : 2.8.1
pip              : 19.3.1
setuptools       : 49.2.0
Cython           : 0.29.21
pytest           : 3.6.4
hypothesis       : None
sphinx           : 1.8.5
blosc            : None
feather          : 0.4.1
xlsxwriter       : None
lxml.etree       : 4.2.6
html5lib         : 1.0.1
pymysql          : None
psycopg2         : 2.7.6.1 (dt dec pq3 ext lo64)
jinja2           : 2.11.2
IPython          : 5.5.0
pandas_datareader: None
bs4              : 4.6.3
bottleneck       : 1.3.2
fsspec           : 0.7.4
fastparquet      : None
gcsfs            : None
matplotlib       : 3.2.2
numexpr          : 2.7.1
odfpy            : None
openpyxl         : 2.5.9
pandas_gbq       : 0.11.0
pyarrow          : 0.14.1
pytables         : None
pyxlsb           : None
s3fs             : 0.4.2
scipy            : 1.4.1
sqlalchemy       : 1.3.18
tables           : 3.4.4
tabulate         : 0.8.7
xarray           : 0.15.1
xlrd             : 1.1.0
xlwt             : 1.3.0
numba            : 0.48.0

</details>

### PR DESCRIPTION
**Pull request by weikhor**
- [x] closes #35517 
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/tests/apply/test_frame_apply.py [hunk 1]
```diff
@@ -1551,3 +1551,11 @@ def foo(x):
...
     with tm.assert_produces_warning(UserWarning, match="Hello, World!"):
         df.agg([foo])
+
+
+def test_apply_return_type():
+    # GH 35517
+    df = DataFrame([["foo"]])
+    result = type(df.apply(lambda col: np.array("bar")).iloc[0])
+    expected = np.ndarray
+    assert result == expected
```
Comment (jreback) [inline pandas/tests/apply/test_frame_apply.py:1561]: can you actually construct the expected exactly and use tm.assert_equal

File: pandas/tests/apply/test_frame_apply.py [hunk 2]
```diff
@@ -1551,3 +1551,11 @@ def foo(x):
     df = DataFrame({"a": [1, 2, 3]})
     with tm.assert_produces_warning(UserWarning, match="Hello, World!"):
         df.agg([foo])
+
+
+def test_apply_return_type():
```
Comment (jreback) [inline pandas/tests/apply/test_frame_apply.py:1556]: there are other tests for return_type can you co-locate

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/apply/test_frame_apply.py (hunk 1 above)
Comment (weikhor) [reply pandas/tests/apply/test_frame_apply.py]: Hi @jreback 
I think I cannot do it using tm.assert_equal. The expected is `np.ndarray` type, not `DataFrame` type.

File: pandas/tests/apply/test_frame_apply.py (hunk 2 above)
Comment (weikhor) [reply pandas/tests/apply/test_frame_apply.py]: Have relocated the file to `pandas/tests/frame/methods/test_astype.py`

File: pandas/tests/frame/methods/test_dtypes.py [hunk 3]
```diff
@@ -131,3 +131,9 @@ def test_dtypes_timedeltas(self):
             index=list("ABCD"),
         )
         tm.assert_series_equal(result, expected)
+
+    def test_frame_apply_np_array_return_type(self):
+        # GH 35517
+        df = DataFrame([["foo"]])
+        df = df.apply(lambda col: np.array("bar")).iloc[0]
```
Comment (jreback) [inline pandas/tests/frame/methods/test_dtypes.py:138]: pls actually construct the expected value and use `tm.assert_frame_equal`

File: pandas/tests/apply/test_frame_apply.py (hunk 1 above)
Comment (mroeschke) [reply pandas/tests/apply/test_frame_apply.py]: Could you assert the full result of `df.apply(lambda col: np.array("bar"))`? Not including the the `iloc[0]` part or `type`

Comment (weikhor) [reply pandas/tests/apply/test_frame_apply.py]: Yes. I have remove iloc[0] part. Using `tm.assert_series_equal` in test case.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks again @weikhor

**Pull request merged**